src/AutoSuggest.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
+
+import { Person } from './types/Person';
+import { peopleFromServer } from './data/people';
+
+type Props = {
+  delay?: number;
+  onSelected: (person: Person | null) => void;
+};
+
+export const AutoSuggest: React.FC<Props> = ({
+  delay,
+  onSelected,
+}) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isInputFocused, setIsInputFocused] = useState(false);
+
+  const searchPerson = query.trim().toLowerCase();
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(person => (
+      person.name.toLowerCase().includes(searchPerson)));
+  }, [peopleFromServer, appliedQuery]);
+
+  const applyQuery = useCallback(debounce((value: string) => {
+    setAppliedQuery(value);
```

Comment:
Debounce function is not implemented correctly. You need to call the debounced function from within your event handler after creating it outside the scope of the event handler.